media-libs/kodi-platform: version bump 17.1.0.1, add live 9999 ebuild

candrews opened the pull request:
https://bugs.gentoo.org/show_bug.cgi?id=603278

Thanks to @herrnst for getting this started in his overlay.

candrews commented:
@SoapGentoo pinging as usual :)

candrews commented:
@fordfrog how's this PR look? (Pinging you as you're maintainer)

fordfrog commented:
sorry for delay. i was trying to push kodi somewhere when i was attempting to compile gentoo on rpi2. since then i dropped that idea.

anyway, i did not have time to test the ebuilds but i have some questions:
1. is it really so that kodi-platform can compile and run against any version of kodi? in my ebuilds i used something like this to make sure that the versions are not different: =media-tv/kodi-16*
2. is it really so that media-tv/kodi and dev-libs/libplatform are not needed during runtime (they are missing from RDEPEND in your ebuilds)?

candrews commented:
@fordfrog I fixed the RDEPEND issue. I also set the dependency on kodi as you indicated (I asked on the #kodi IRC channel, and they confirmed your thought that kodi and kodi-platform major versions must match). Finally, because >=kodi-17 is hard masked, I added a hard mask on >=kodi-platform-17.

For kodi-platform-9999, I didn't specify a kodi version dependency... which I think is appropriate.

How's it look now?

fordfrog commented:
@candrews thank you for confirming my assumption. the ebuilds look better now, just the dep in version 9999 should imo depend on =media-tv/kodi-9999*, otherwise it would be possible to attempt to compile kodi-platform-9999 against kodi-16 for example which might not work and is not how it is meant to work.

candrews commented:
As I said, I was on the fence :-) I've made that change.

fordfrog commented:
@candrews perfect! now the last thing that should be done (it has changed overnight) is removing the ebuild from package.mask as ffmpeg-3 has been removed from package.mask. after that, i think, the patch can be merged :-)

candrews commented:
@fordfrog but the Kodi hard mask remains, and since kodi-platform and Kodi need to be the same version, they both must be hard masked or neither hard masked.

https://github.com/gentoo/gentoo/pull/3308

So unless you'll remove the Kodi hard mask, I think things have to stay the way they are in this PR.

candrews commented:
@fordfrog the hard mask was removed - this is ready for merging.

SoapGentoo commented on media-libs/kodi-platform/kodi-platform-17.1.0.1.ebuild:
```diff
@@ -0,0 +1,36 @@
+	SRC_URI="https://github.com/xbmc/kodi-platform/tarball/${GIT_COMMIT} -> ${P}.tar.gz"
+	S="${WORKDIR}/xbmc-kodi-platform-${GIT_COMMIT}"
+	;;
+esac
```
could you change this to
```
if [[ ${PV} == *9999 ]]; then
	EGIT_REPO_URI="git://github.com/xbmc/kodi-platform.git"
	inherit git-r3
else
	KEYWORDS="~amd64 ~x86"
	GIT_COMMIT="c8188d8"
	SRC_URI="https://github.com/xbmc/kodi-platform/tarball/${GIT_COMMIT} -> ${P}.tar.gz"
	S="${WORKDIR}/xbmc-kodi-platform-${GIT_COMMIT}"
fi
```
which is more common in the Gentoo-wild

fordfrog commented:
merged in dcbd61831d00f7d2254f50f4e8fafeed2a90217f. thanks!
